scripts/weihergames_data_extractor.py

```diff
@@ -0,0 +1,319 @@
+            self.__img = getRelativeImagePath(_ALT_PLAYER_IMG)
+
+class GameData():
+    def __init__(self, name):
+        self.name = name
+        self.img = None
+        self.field_of_play = ''
+        self.field_of_play_icon = None
+        self.pointrules = []
+        self.specific_leveling_value = 0
+    def __str__(self):
+        return 'GameData for {0}'.format(self.name)
+
+    @property
+    def img(self):
+        return self.__img
+
+    @img.setter
+    def img(self, img_filename):
+        img_path = os.path.join(_IMAGE_FOLDER, str(img_filename))
+        if os.path.isfile(img_path):
+            self.__img = getRelativeImagePath(img_filename)
+        else:
+            self.__img = getRelativeImagePath(_ALT_GAME_IMG)
+
+    @property
+    def field_of_play_icon(self):
+        return self.__field_of_play_icon
+
+    @img.setter
+    def field_of_play_icon(self, img_filename):
+        img_path = os.path.join(_IMAGE_FOLDER, str(img_filename))
+        if os.path.isfile(img_path):
+            self.__field_of_play_icon = getRelativeImagePath(img_filename)
+        else:
+            self.__field_of_play_icon = getRelativeImagePath(_ALT_FIELDOFPLAY_IMG)
+
+
+class MatchData():
+    def __init__(self, game):
+        self.game = game
+        self.participants= []
+        self.elos_after = [] # must have same oder as participants
+        self.elos_delta = [] # must have same oder as participants
+        self.result = [] # must have same oder as participants
+        self.datetime = ''
+        self.tie = False
+        self.location = ''
+        self.remarks = ''
+    def __str__(self):
+        return 'MatchData of [{0}] played by: {1}'.format(self.game,self.participants)
+
+    @property
+    def winner(self):
+        if 'lowerwins' in self.game.pointrules:
+            # LOWEST SCORE WINS:
+            search_value = min(self.result)
+        else:
+            # HIGEHST SCORE WINS:
+            search_value = max(self.result)
+        # get index of result array with the highest or lowest result:
+        index_of = [i for i, j in enumerate(self.result) if j == search_value]
+        if len(index_of) > 1:
+            self.tie = True
+            return [self.participants[x] for x in index_of]
+        elif len(index_of) == 1:
+            return self.participants[index_of[0]]
+        else:
+            logging.error('NO WINNER IDENTIFIED ON: {0}'.format(self))
+            return None
+
+
+
+
+
+
+def read_player_from_json():
+    players = []
+    # Load data
+    j = json.load(open(JSON_DATABASE))
+    # Sanity check of json file:
+    pass
+    #output all warnings + errors, if no error continue:
+    pass
+    # set player:
+    for name, data in j['members'].items():
+        p = PlayerData(name)
+        p.elo = ELO_START_VALUE
+        p.img = data.get('img', None)
+        players.append(p)
+        logging.debug('Player found in DB: {0}'.format(p))
+    return players
+
+
+def read_games_from_json():
```

Comment:
![image](https://user-images.githubusercontent.com/29222809/80313373-dd1d3f80-87ea-11ea-99b8-b3425c41df16.png)

Only assign match to data where the player iself attended

Reply:
The player (doompy) did not attend crossboccia in this screenshot